<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Robot path planning</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="SceneWidget" name="sceneWidget">
     <property name="minimumSize">
      <size>
       <width>600</width>
       <height>0</height>
      </size>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="controlWidget" native="true">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="LineChartWidget" name="graphWidget" native="true">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>150</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QListWidget" name="waypointWidget">
        <property name="viewMode">
         <enum>QListView::ViewMode::ListMode</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="jointInputWidget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QPushButton" name="prevButton">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>20</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>&lt;</string>
           </property>
           <property name="flat">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="j0Field">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="minimum">
            <number>-360</number>
           </property>
           <property name="maximum">
            <number>360</number>
           </property>
           <property name="singleStep">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="j1Field">
           <property name="minimum">
            <number>-360</number>
           </property>
           <property name="maximum">
            <number>360</number>
           </property>
           <property name="singleStep">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="j2Field">
           <property name="minimum">
            <number>-360</number>
           </property>
           <property name="maximum">
            <number>360</number>
           </property>
           <property name="singleStep">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="j3Field">
           <property name="minimum">
            <number>-360</number>
           </property>
           <property name="maximum">
            <number>360</number>
           </property>
           <property name="singleStep">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="j4Field">
           <property name="minimum">
            <number>-360</number>
           </property>
           <property name="maximum">
            <number>360</number>
           </property>
           <property name="singleStep">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="j5Field">
           <property name="minimum">
            <number>-360</number>
           </property>
           <property name="maximum">
            <number>360</number>
           </property>
           <property name="singleStep">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="nextButton">
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>20</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>&gt;</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="trajectoryWidget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QPushButton" name="waypointButton">
           <property name="text">
            <string>New Waypoint</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="startIdxField"/>
         </item>
         <item>
          <widget class="QSpinBox" name="endIdxField">
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="trajectoryButton">
           <property name="text">
            <string>Evaluate Trajectory</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="fileIOWidget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QPushButton" name="serializeButton">
           <property name="text">
            <string>Serialize</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="deserializeButton">
           <property name="text">
            <string>Deserialize</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SceneWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>SceneWidget.h</header>
  </customwidget>
  <customwidget>
   <class>LineChartWidget</class>
   <extends>QWidget</extends>
   <header>LineChartWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
